dnf5/context.cpp

```diff
@@ -352,20 +333,69 @@ class PkgDownloadCB : public libdnf::repo::DownloadCallbacks {
     std::string what;
 };
 
-std::chrono::time_point<std::chrono::steady_clock> PkgDownloadCB::prev_print_time = std::chrono::steady_clock::now();
+std::chrono::time_point<std::chrono::steady_clock> DownloadCB::prev_print_time = std::chrono::steady_clock::now();
 
 }  // namespace
 
+static bool is_url(std::string path) {
+    for (auto & ch : path) {
+        if (ch != ':') {
```

Comment:
I'd still like a proper url-parsing function much more, but since you're doing it in this simple way, how about checking also for `ch != '/'`, so that you don't iterate over the whole path in case it's not a url?

Reply:
OK. I added check for `'/'`.